src/main/java/com/intuit/graphql/orchestrator/batch/AuthDownstreamQueryModifier.java

```diff
@@ -112,8 +120,10 @@ public TraversalControl visitField(Field node, TraverserContext<Node> context) {
       GraphQLFieldDefinition fieldDefinition = getFieldDefinition(node.getName(), parentType);
       requireNonNull(fieldDefinition, "Failed to get Field Definition for " + node.getName());
 
-      if (serviceMetadata.shouldModifyDownStreamQuery() && (hasResolverDirective(fieldDefinition)
-          || isExternalField(parentType.getName(), node.getName()))) {
+      boolean shouldRemoveNode = (serviceMetadata.shouldModifyDownStreamQuery() && (hasResolverDirective(fieldDefinition)
+              || isExternalField(parentType.getName(), node.getName())))
+              || shouldIgnoreNode(node, this.queryVariables);
+      if (shouldRemoveNode) {
```

Comment:
I think this will remove the field but the variable definition that is refered to by @skip ad @sinclude might still be sent to downstream as unreferenced?
We need to revisit.. might want to check the solution https://github.com/graph-quilt/graphql-orchestrator-java/issues/169 but I'm open to an alternative.